akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/ActorSystemStub.scala

```diff
@@ -53,8 +53,8 @@ import scala.concurrent._
     terminationPromise.trySuccess(Terminated(this)(null))
     terminationPromise.future
   }
-  override def whenTerminated: Future[akka.actor.typed.Terminated] = terminationPromise.future
-  override def getWhenTerminated: CompletionStage[Terminated] = FutureConverters.toJava(whenTerminated)
+  override def whenTerminated: Future[Done] = terminationPromise.future.map(_ ⇒ Done)
```

Comment:
This will make a new lambda/transform for every call, I guess we could cache a single transformed Future in a field and always return that. Not sure it matters though.